### PR TITLE
Cosmosis standard library

### PR DESCRIPTION
**Pull request by vitenti**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by vitenti** [comment]
@conda-forge-admin, please lint

File: recipes/cosmosis-standard-library/build.sh
Comment (beckermr) [inline recipes/cosmosis-standard-library/build.sh:25]: We have a policy of removing all `*.a` files (ie static libs) and making special packages for them. Is there a reason this is commented out? 

This might relate to some of our previous conversations. I don't recall and my apologies if it does!

File: recipes/cosmosis-standard-library/meta.yaml
Comment (beckermr) [inline recipes/cosmosis-standard-library/meta.yaml:15]: The default here is true and so this line should not be needed.

File: recipes/cosmosis-standard-library/meta.yaml
Comment (beckermr) [inline recipes/cosmosis-standard-library/meta.yaml:17]: What's going on here?

File: recipes/cosmosis-standard-library/meta.yaml
Comment (beckermr) [inline recipes/cosmosis-standard-library/meta.yaml:32]: These mpi selectors don't do anything here since they are only evaluated at run time. Please see the conda-forge docs for how to build MPI variants for your packages. https://conda-forge.org/docs/maintainer/knowledge_base.html#message-passing-interface-mpi

File: recipes/cosmosis-standard-library/meta.yaml
Comment (beckermr) [inline recipes/cosmosis-standard-library/meta.yaml:46]: This host section cannot be right. Only things that cosmosis links to via code need to be here. For numpy specifically, see the docs on how to link to numpy properly. https://conda-forge.org/docs/maintainer/knowledge_base.html#building-against-numpy

File: recipes/cosmosis-standard-library/meta.yaml
Comment (beckermr) [inline recipes/cosmosis-standard-library/meta.yaml:67]: The license file here is incorrect. We cannot simply apply random licenses to other people's code and then distribute it on conda-forge. As far as I can tell, the upstream package has no license. @joezuntz will need to figure out the correct license for the upstream code and apply the license himself there before we can merge, this.

File: recipes/cosmosis-standard-library/meta.yaml
Comment (beckermr) [inline recipes/cosmosis-standard-library/meta.yaml:73]: All maintainers need to comment saying they are ok to maintain the recipe.

**Review by beckermr — CHANGES_REQUESTED** [review]

File: recipes/cosmosis-standard-library/meta.yaml
Comment (joezuntz) [reply recipes/cosmosis-standard-library/meta.yaml:67]: This is very complicated, and is one reason I have avoided trying to share a compiled version of the CSL.

The different bits of included code are under different licenses. Some are standard licenses, but there are many different ones, GPL, LGPL, BSD, MIT. 

Some are vague, like  "You can use X freely, provided that in your publications, you cite at least the paper ..." or similar generic statements.

I understand it is permitted or acceptable to share the source code of all of them, but I would not recommend trying to sort through this to the level needed here.

There is already a pacakge `cosmosis-build-standard-library` which downloads and builds the CSL for the user. I would strongly recommend using that instead, especially as most users of CSL are also developers.  It supplies a one-line command to download and build the CSL.  If more is needed then we could add something to that command to install the result to a central location.

**Pull request closed without merging**
